pkg/api/apikey.go

```diff
@@ -75,13 +75,17 @@ func (hs *HTTPServer) AddAPIKey(c *models.ReqContext, cmd models.AddApiKeyComman
 	var err error
 	var serviceAccount *models.User = &models.User{Id: -1}
 	if hs.Cfg.FeatureToggles["service-accounts"] {
+		//Every new API key must have an associated service account
 		if cmd.CreateNewServiceAccount {
+			//Create a new service account for the new API key
 			serviceAccount, err = hs.SQLStore.CloneUserToServiceAccount(c.Req.Context(), c.SignedInUser)
 			if err != nil {
 				return response.Error(500, "Unable to clone user to service account", err)
 			}
 			cmd.ServiceAccountId = serviceAccount.Id
 		} else {
+			//Link the new API key to an existing service account
+
```

Comment:
now it makes more sense.

The branching inside here could get messy, this could potentially be refactored in the future if it gets mroe complicated.
But this is great as a first step!

Reply:
I think this will be repeatedly refactored, but this is the smallest change I could make that did the job, for now.